src/wallet/wallet.cpp

```diff
@@ -3,8 +3,6 @@
 // Distributed under the MIT software license, see the accompanying
 // file COPYING or http://www.opensource.org/licenses/mit-license.php.
 
-#include <wallet/wallet.h>
```

Comment:
Is this intentional? Could this cause the current travis issue?

Reply:
@jonasschnelli I moved it to `wallet.h` because OutputType is needed in the header. My C++ is a bit rusty; do I need to keep it in the cpp file? My local machine doesn't care.

Reply:
Perhaps a forward declaration would suffice?

Reply:
I think this is likely the travis issue.

Has a similar error in Elements project that was exactly including `wallet.h` in a new file, we had to def guard it, may not be appropriate here.

Reply:
@jb55 sadly `ISO C++ forbids forward references to 'enum' types`.

Reply:
> @jb55 sadly ISO C++ forbids forward references to 'enum' types.

This is only true for enums that don't specify a width. If you define an enum like `enum MyEnum : int {...};`, you can forward declare it with `enum MyEnum : int;`. So you could add `: int` to the `OutputType` definition in wallet.h in order to be able to forward declare it other places. Moving it seems fine, too, though.